docs/SPEC.md

```diff
@@ -569,13 +531,13 @@ Values must also be of the same type.
 ##### Function types
 
 A _function type_ represents a set of all functions with the same argument and result types.
+Functions arguments are always named. (There are no positional arguments.)
+Therefore implementing a function type requires that the arguments be named the same.
 
-##### Generator types
-
-A _generator type_ represents a value that produces an unknown number of other values.
-The generated values may be of any other type but must all be the same type.
+##### Stream types
 
-<!-- no corresponding issue [IMPL#658](https://github.com/influxdata/platform/issues/658) Implement Generators types -->
+A _stream type_ represents an unbounded collection of values.
+The values may be of any other type but must all be the same type.
```

Comment:
What does "any other" mean here? Does it mean there can't be nested stream types?